File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/FizzBuzzValueCommand.java
package fizzbuzz;

public class FizzBuzzValueCommand implements IFizzBuzzCommand {
  private FizzBuzzType _type;

  public FizzBuzzValueCommand(FizzBuzzType type) {
    _type = type;
  }

  @Override
  public IFizzBuzz execute(int number) {
    return _type.generate(number);
  }

}

File: dev/20190902/k2works/app/FizzBuzz/Domain/FizzBuzzValueCommand.cs
namespace FizzBuzz.Domain
{
    public class FizzBuzzValueCommand: IFizzBuzzCommand
    {
        private FizzBuzzType _type;

        public FizzBuzzValueCommand(FizzBuzzType type)
        {
            _type = type;
        }
        public IFizzBuzz Execute(int number)
        {
            return _type.Generate(number);
        }
    }
}